src/parse.c

```diff
@@ -2108,6 +2108,33 @@ handle_vrf_interfaces(NetplanParser* npp, yaml_node_t* node, const void* data, G
     return TRUE;
 }
 
+static gboolean
+handle_vxlan_source_port(NetplanParser* npp, yaml_node_t* node, const void* _, GError** error)
+{
+    assert_type(npp, node, YAML_SEQUENCE_NODE);
+    if (node->data.sequence.items.top - node->data.sequence.items.start != 2)
+        return yaml_error(npp, node, error, "%s: Expected exactly two values for port-range",
+                          npp->current.netdef->id);
```

Comment:
question: how common is it to want to specifiy a single port?

I've never used VXLAN, but is it at all common to specify a range of 1? If so, we might want to accept scalar values? I'm guessing the answer is "not that common" given what I'm seeing from cursory googling, but I still think the question merits asking.

Reply:
I'm not an expert on VXLAN myself and agree with your "not that common" assessment. Therefore, I'll leave this as-is.
It can still be defined as e.g.: `port-range: [4200, 4200]`. If it turns out to be a common case, we could easily implement this usability optimization, but should also be reconsidering the "port-range" settings name in this case.